lib/spack/spack/binary_distribution.py

```diff
@@ -646,7 +648,8 @@ def get_specs(force=False):
                 # we need to mark this spec concrete on read-in.
                 spec = Spec.from_yaml(f)
                 spec._mark_concrete()
-                _cached_specs.append(spec)
+                if spec.satisfies(arch_spec):
```

Comment:
@becker33 For installing macOS binaries using a linux host this restriction needs to be removed. Only the change at line 630 needs to be made.

Reply:
` git diff upstream/develop
diff --git a/lib/spack/spack/binary_distribution.py b/lib/spack/spack/binary_distribution.py
index cab0ad2..a4624e9 100644
--- a/lib/spack/spack/binary_distribution.py
+++ b/lib/spack/spack/binary_distribution.py
@@ -609,7 +609,6 @@ def get_specs(force=False):
         tty.warn("No Spack mirrors are currently configured")
         return {}
 
-    path = str(spack.architecture.sys_type())
     urls = set()
     for mirror_name, mirror_url in mirrors.items():
         if mirror_url.startswith('file'):
@@ -626,7 +625,7 @@ def get_specs(force=False):
             tty.msg("Finding buildcaches on %s" % mirror_url)
             p, links = spider(mirror_url + "/" + _build_cache_relative_path)
             for link in links:
-                if re.search("spec.yaml", link) and re.search(path, link):
+                if re.search("spec.yaml", link):
                     urls.add(link)
 
     _cached_specs = []
`